table.go

```diff
@@ -1115,55 +1115,88 @@ func (t *TableBlock) abort(granule *Granule) {
 	}
 }
 
-func (t *TableBlock) Serialize() ([]byte, error) {
+func (t *TableBlock) Serialize(writer io.Writer) error {
 	ctx := context.Background()
 
 	// Read all row groups
 	rowGroups := []dynparquet.DynamicRowGroup{}
-	// math.MaxUint64 is passed because we want to serialize everything.
+
+	// Collect all the row groups just to determine the dynamic cols
 	err := t.RowGroupIterator(ctx, math.MaxUint64, nil, &AlwaysTrueFilter{}, func(rg dynparquet.DynamicRowGroup) bool {
 		rowGroups = append(rowGroups, rg)
 		return true
 	})
 	if err != nil {
-		return nil, err
+		return err
 	}
-
 	merged, err := t.table.config.schema.MergeDynamicRowGroups(rowGroups)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	buf := bytes.NewBuffer(nil)
 	cols := merged.DynamicColumns()
-	w, err := t.table.config.schema.GetWriter(buf, cols)
+	w, err := t.table.config.schema.GetWriter(writer, cols)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer t.table.config.schema.PutWriter(w)
+	defer w.Close()
+
+	// Iterate over all the row groups, and write them to storage
+	count := 0
+	maxRows := 8096
```

Comment:
I assume you meant 8192 (2pow13)

Reply:
lmao I have refactored enough times that it somewhere got typed wrong